xlml/utils/tpu.py

```diff
@@ -372,3 +386,38 @@ def clean_up_idle_queued_resources(project_name: str, zones: Iterable[str]) -> N
       ):
         logging.info(f'Deleting {qr.name} in {state.name} status.')
         client.delete_queued_resource(name=qr.name)
+
+
+@task
+def clean_up_idle_nodes(project_name: str, zones: Iterable[str]) -> None:
+  """Clean up TPU nodes that are expired.
+
+  Args:
+   project_name: The project of resources.
+   zones: Available zones to clean up for the project.
+  """
+  creds, _ = google.auth.default()
+  client = tpu_api.TpuClient(credentials=creds)
+
+  logging.info(f'Cleaning up nodes in project {project_name}.')
+  for zone in zones:
+    logging.info(f'Checking in zone {zone.value}.')
+    parent = f'projects/{project_name}/locations/{zone.value}'
+    request = tpu_api.types.ListNodesRequest(parent=parent)
+    responses = client.list_nodes(request)
+
+    for node in responses:
+      create_time = node.create_time
+      current_time = datetime.datetime.now(datetime.timezone.utc)
+      ttl = int(node.labels[TTL]) if node.labels[TTL] else 0
+      if ttl != 0:
+        create_time = node.create_time
+        current_time = datetime.datetime.now(datetime.timezone.utc)
```

Comment:
`create_time` and `current_time` are also defined above.

Reply:
oops, will remove it.

Reply:
Done.